draft-ietf-tls-tls13.md

```diff
@@ -4375,6 +4375,91 @@ currently deployed and properly configured TLS implementations.
 %%### Ticket Establishment
 %%### Updating Keys
 
+# State Machine
+
+This section provides a summary of the legal state transitions for the
+client and server handshakes.  State names (in all capitals, e.g.,
+START) have no formal meaning but are provided for ease of
+comprehension.  Messages which are sent only sometimes are indicated
+in [].
+
+## Client
+
+~~~~
+                  +-----> START
+             Recv |        |
+            Hello |        | Send ClientHello
+            Retry |        v
+        +-> Request +---- WAIT_SH
+    Can |                    | Recv ServerHello
+   Send |                    V
+  Early |                 WAIT_EE
+   Data |                    | Recv EncryptedExtensions
+        |           +--------+--------+
+        |     Using |                 | Using Certificate
+        |       PSK |                 v
+        |           |            WAIT_CERT_CR
+        |           |        Recv |       | Recv CertificateRequest
+        |           | Certificate |       v
+        |           |             |    WAIT_CERT
+        |           |             |       | Recv Certificate
+        |           |             v       v
+        |           |              WAIT_CV
+        |           |                 | Recv CertificateVerify
+        |           +> WAIT_FINISHED <+
+        |                  | Recv Finished
+        |                  |
+        +->                | [Send EndOfEarlyData]
+                           | [Send Certificate [+ CertificateVerify]]
+                           | Send Finished
+ Can Send                  v
+ App Data -->          CONNECTED
+ After
+ Here
+~~~~
+
+## Server
+
+~~~~
+                      +----> START
+                 Send |        |
+                Hello |        | Recv ClientHello
+                Retry |        |
+              Request |        v
+                      +--- RECVD_CH
+                               | Select parameters
+                               v
+                            NEGOTIATED
+                               | Send ServerHello
+                               | Send EncryptedExtensions
+                               | [Send CertificateRequest]
+Can Send                       | [Send Certificate + CertificateVerify]
+App Data -->                   | Send Finished
+After                 +--------+--------+
+Here            0-RTT |                 | 0-RTT
+                      |                 v
+                      |             WAIT_EOED <---+
+                      |        Recv |       |     | Recv
+                      |        EOED |       |     | Early Data
+                      |             |       +-----+
```

Comment:
EOED is still abbreviated. I feel like ideally we'd also have that be the "core" branch while the WAIT_EOED self-loop is more a side thing. That also solves our abbreviation problem:

```
Can Send                       | [Send Certificate + CertificateVerify]
App Data -->                   | Send Finished
After                 +--------+--------+
Here         No 0-RTT |                 | 0-RTT
                      |                 v
                      |             WAIT_EOED ---+
                      |            Recv |  ^     | Recv Early Data
                      |  EndOfEarlyData |  |     | 
                      |                 |  +-----+
                      +--> WAIT_FLT2 <--+
                               |
```